interfaces/builtin/mir.go

```diff
@@ -0,0 +1,606 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (c) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more dtails.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.
+ *
+ */
+
+package builtin
+
+import (
+       "github.com/snapcore/snapd/interfaces"
+)
+
+var mirPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Mir server. Reserved because this
+# gives privileged access to the system.
+# Usage: reserved
+
+
+capability dac_override,
+capability sys_tty_config,
+capability sys_admin,
+
+unix (receive, send) type=seqpacket addr=none,
+/tmp/mir_socket rw,
+/dev/dri/card0 rw,
+/dev/shm/\#* rw,
+
+/sys/devices/**/uevent rw,
+/sys/devices/**/ r,
+/dev/input/* rw,
+/dev/tty* wr,
+/run/udev/data/* r,
+/run/udev/** rw,
+
+ptrace peer=**,
```

Comment:
If mir server needs ptrace, it should be more fine-grained than this and only ptrace for connected clients. Please use something like this in mirConnectedPlugAppArmor:
  ptrace (tracedby) peer=###SLOT_SECURITY_TAGS###,

Then the mirConnectedSlotAppArmor would have:
  ptrace (trace) peer=###PLUG_SECURITY_TAGS###,

The above assumes you need 'trace' Can you paste the apparmor denial that prompted this rule?